### ISSUE TITLE
Could not run the testcase build_encrypted_with_passphrase of e2e-test

### ISSUE THREAD

**Issue by sailer1205** [apptainer/singularity#5342]
### Version of Singularity:

What version of Singularity are you using? Run:

```
$ singularity version
3.6.0-rc.2+46-g971f9116b
```

### Expected behavior

run the testcase build_encrypted_with_passphrase of e2e-test successfully

### Actual behavior

it went wrong, the error message is:

> --- FAIL: TestE2E/PAR/BUILD/build_encrypted_with_passphrase (0.09s)
>     singularitycmd.go:663: Running command "/usr/local/bin/singularity sif info 2 /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_cmd
>     singularitycmd.go:663: 
>         "/usr/local/bin/singularity sif info 2 /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_cmdline_option.sif" output:
>         Error for command "info": opening(RDONLY) container file: open /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_cmdline_option
>         
>         Usage:
>           singularity [global options...] sif info <descriptorid> <containerfile>
>         
>         Run 'singularity --help' for more detailed usage information.
>         
>     singularitycmd.go:663: got 1 as exit code and was expecting 0: exit status 1
>         waiting for command "/usr/local/bin/singularity sif info 2 /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_cmdline_option.sif
>         github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func1
>         	github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:638
>         github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func2
>         	github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:663
>         runtime.goexit
>         	runtime/asm_amd64.s:1357
>     singularitycmd.go:663: Running command "/usr/local/bin/singularity sif info 2 /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_cmd
>     singularitycmd.go:663: Running command "/usr/local/bin/singularity sif info 2 /tmp/stest.441373290/passphrase-encryption.279525314/encrypted_env
>     imgbuild.go:36: Test TestE2E/PAR/BUILD/build_encrypted_with_passphrase failed, not removing /tmp/stest.441373290/passphrase-encryption.279525314

And I found that the encrypted_cmdline_option.sif was not generated.
Is there something wrong in the testcase?
the source code of the testcase is below:
https://github.com/hpcng/singularity/blob/4f0b570d58883190ba2e4bd3b565882cb1274651/e2e/imgbuild/imgbuild.go#L893

### Steps to reproduce this behavior

$SINGULARITY_E2E=1 make -C builddir e2e-test

### What OS/distro are you running

```
$ cat /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.1 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.1"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.1 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.1:GA"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.1
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.1"

```

### How did you install Singularity

source.

**Comment by dtrudg** [comment]
Are you able to provide a full log of the test case execution?

RHEL8 has an issue with mksquashfs that can result in occasional failure to create a container under heavy load, and this can occur in the e2e tests which run a lot of mksquashfs, sometimes in parallel.

If it is this issue - it has been reported upstream, but no response to date:

https://bugzilla.redhat.com/show_bug.cgi?id=1785652

If it's something else, we'll look into it.

**Comment by sailer1205** [comment]
@dctrud 

> Are you able to provide a full log of the test case execution?

this is the full log for the testcase execution.
> 
> RHEL8 has an issue with mksquashfs that can result in occasional failure to create a container under heavy load, and this can occur in the e2e tests which run a lot of mksquashfs, sometimes in parallel.
> 
> If it is this issue - it has been reported upstream, but no response to date:
> 
> https://bugzilla.redhat.com/show_bug.cgi?id=1785652

I think this is not a mksquashfs bug. When I deleted a line of the code, it was executed successfully:
```
        c.env.RunSingularity(
                t,
                // e2e.AsSubtest("passphrase flag"),
                e2e.WithProfile(e2e.RootProfile),
                e2e.WithCommand("build"),
                e2e.WithArgs(cmdArgs...),
                e2e.ConsoleRun(passphraseInput...),
                e2e.ExpectExit(expectedExitCode),
        )
```
it's very strange!  the method AsSubtest just appends a subtitle for the testcase, why dose it effect the result of the testcase execution?

> 
> If it's something else, we'll look into it.

**Comment by dtrudg** [comment]
@sailer1205 - is this repeatable? I.e. does it always fail without commenting out, and always succeed with commenting out?

We need to know if this is a repeatable issue, or something that is occasional.

**Comment by sailer1205** [comment]
@dctrud 
Yes, it's repeatable. I also do the test on centos7, it has the same result.

**Comment by sailer1205** [comment]
@dctrud 
I found the reason, this may be a golang's bug.
In my environment, I got the go binary from the source, the src path is /home/test/go
and the branch is release-branch.go1.14.
The go binary is under /home/test/go/bin and I set the go environment by:

> export GOROOT=/home/test/go
> export PATH=$GOROOT/bin:$PATH

Under this environment, the error was reproduced.

But when I move the go src into /usr/local directory and set the go environment by:

> export PATH=/usr/local/go/bin

and recompile the singularity,  then use the new singularity binary to run the e2e test,
the case was passed!
Should we only put the go under the system path?
